File: Solutions/2. Add Two Numbers/2. Add Two Numbers.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
    head:=&ListNode{Val:0}
    extra:=0
    curr:=head
    for l1!=nil || l2!=nil || extra!=0{
        if l1!=nil{
            extra+=l1.Val
            l1=l1.Next
        }
        if l2!=nil{
            extra+=l2.Val
            l2=l2.Next
        }
        curr.Next=&ListNode{Val:extra%10}
        curr=curr.Next
        extra/=10
    }
    return head.Next
}

File: Solutions/2. Add Two Numbers/2. Add Two Numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* prehead=new ListNode(0);
        ListNode* p=prehead;
        int extra=0;
        while(l1 or l2 or extra){
            if(l1){
                extra+=l1->val;
                l1=l1->next;
            }
            if(l2){
                extra+=l2->val;
                l2=l2->next;
            }
            p->next=new ListNode(extra%10);
            p=p->next;
            extra/=10;
        }
        p->next=NULL;
        return prehead->next;
    }
};